streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -338,6 +370,48 @@ abstract class DStream[T: ClassTag] (
   }
 
   /**
+   * Helper method to generate an RDD for the specified time.
+   * This sets and resets the relevant local variables before and after the call to compute.
+   */
+  private def doCompute(time: Time): Option[RDD[T]] = {
+    val scopeKey = SparkContext.RDD_SCOPE_KEY
+    val scopeNoOverrideKey = SparkContext.RDD_SCOPE_NO_OVERRIDE_KEY
+    // Pass this DStream's operation scope and creation site information to RDDs through
+    // thread-local properties in our SparkContext. Since this method may be called from another
+    // DStream, we need to temporarily store any old scope and creation site information to
+    // restore them later after setting our own.
+    // TODO: this won't work if multiple StreamingContexts share the same SparkContext
```

Comment:
Why wont this work? Each streaming context will be calling compute in its own scheduling thread. They are not going to interfere with each other.